scout/server/blueprints/panels/views.py

```diff
@@ -37,7 +48,34 @@ def api_panels(panel_name):
 @templated("panels/panels.html")
 def panels():
     """Show all panels for a user"""
-    if request.method == "POST":  # Edit/create a new panel and redirect to its page
+
+    # Add search box and add results if applicable
+    panels_found_compiled = []
+    search_string = ""
+    if request.method == "POST" and request.form.get("search_for"):
+        # Query db for panels containing the search string
+        # TODO: removed escape of form input, sonarcloud will complain
+        hgnc_symbols = ""
+        search_string = escape(request.form.get("search_for"))
+        try:
+            hgnc_symbols = parse_raw_gene_ids([search_string])
+        except ValueError:
+            flash(
+                "Provided gene info could not be parsed! " "Please allow autocompletion to finish.",
+                "warning",
+            )
+        # TODO: dont crash after error
+        LOG.debug("SEARCH: {}".format(search_string))
+        LOG.debug("HGNC: {}".format(hgnc_symbols))
+
+        for hgnc_id in hgnc_symbols:
+            panels_found = store.search_panels(hgnc_id)
+            panels_found_compiled.extend(_compile_panel_versions(panels_found))
```

Comment:
You don't need this if you optimize the query!

Reply:
I don't understand which line you are referring to?

Reply:
I mean that is you optimize the query you don't need _compile_panel_versions and _compile_panel_versions_aux, the adapter's search_panels function can return the list of panel in the right format

Reply:
Ok.

But I assume that the a for-look is still necessary as `parse_raw_gene_symbols(raw_symbols_list)` may return a set with several entries?


```
def parse_raw_gene_symbols(raw_symbols_list):
    """Parse list of concatenated gene symbol list for hgnc_symbols from Phenomizer.

    Arguments:
        raw_symbols(list(string)) - e.g. ("POT1 | MUTYH", "POT1 | ATXN1 | ATXN7")

    Returns:
        hgnc_symbols(set(string)) - set of (unique) gene symbols without intervening chars
    """

```

Reply:
Yes, I guess so!